addons/hr_expense/models/hr_expense_sheet.py

```diff
@@ -464,6 +464,20 @@ def create(self, vals_list):
         sheets.activity_update()
         return sheets
 
+    def write(self, vals):
+        if 'state' in vals or 'approval_state' in vals:
+            # Avoid user with write access on expense sheet in draft state to bypass the validation process
+            valid_states = {'submit', None}
+            if (not self.user_has_groups('hr_expense.group_hr_expense_manager')
+                and all(self.mapped(lambda e: e.state == 'draft')) and vals.get('state') not in valid_states
+                and vals.get('approval_state') not in valid_states):
```

Comment:
```suggestion
            if (
                not self.user_has_groups('hr_expense.group_hr_expense_manager')
                and any(e.state == 'draft' for e in self) 
                and ('submit' in (vals.get('state'), vals.get('approval_state'))
            ):
```
I was thinking maybe the condition could be rewritten this way (but please compare it with your version @JulienAlardot):

We raise if and only if the following 3 conditions are simultaneously met:
- the user doesn't have privileges;
- any expense sheet is currently in `draft`; and
- any of the new `state` or `approval_state` is `submit`.

Let me know your thoughts when you'll be back :smile:

Reply:
```suggestion
            if (
                    not self.user_has_groups('hr_expense.group_hr_expense_manager')
                    and any(state == 'draft' for state in self.mapped('state'))
                    and (vals.get('state') not in valid_states or vals.get('approval_state') not in valid_states)
            ):
```
- the user doesn't have privileges;
- any expense sheet is currently in draft; and
- **any of the new state or approval_state is not submit (or None if no data for the field is provided)**